### PR TITLE
Feature/team fights

### LINKED ISSUE #1933: [Feature] - FE of Team Battles
### Prerequisites

- [X] I checked to make sure that this feature has not already been filed
- [X] I'm reporting this information to the correct repository
- [X] I understand enough about this issue to complete a comprehensive document

### Describe the feature and its requirements

There has already been an issue created for the back end of team fights as well as a pull request for it.
https://github.com/CryptoBlades/cryptoblades/pull/1906

There has not yet been an issue for the front end component of team fights.  Below is a series of mock up images and descriptions that should direct how the front end should look.  

What the current adventure mode looks like

There should be an option added that allows users to switch between individual and team combats.  Adventure indicates an individual combat and team adventure indicates team combat. There should be an indication to which mode the user is in as indicated by the color of the text. 

The text at the top indicating which adventure mode the user is in should remain regardless of which mode the user is in, but the color of the text should change to reflect it.  

There should be a team fight button that executes the team fight and also indicates how many characters will be fighting as X/Y, where X represents the number of characters that are paired up against an enemy and Y represents the number of characters in their plaza.  

To represent the character in the team battle I used the cards from the garrison, but also included the power and element of the equipped weapon.

There should be icons that are clickable and allow for the individual selection of each enemy for each character and icons that allow for the switching of an enemy for a given character.

When either initially selecting or switching between enemy targets there should be a side panel that opens from the right with the available enemy options.  There should also be a box that appear surrounding an enemy when you hover over it just like the one in traditional combat that surrounds that enemy.

The team fight button should be executeable regardless of having a full team or not matched with enemies. 

The team fight result modal should be very similar to that of the individual fight modal.  The main differences include the total amount of unclaimed SKILL/VALOR is presented, individual fight rolls and XP gained is showed per character.

### Is your feature request related to an existing issue? Please describe.

Yes the backend was done in 1905
https://github.com/CryptoBlades/cryptoblades/issues/1905

The pull request is needing to be reviewed at the time of creation
https://github.com/CryptoBlades/cryptoblades/pull/1906

### Is there anything stopping this feature being completed?

N/A

### Describe alternatives you've considered

N/A

### Additional context

I put together the mock images trying to use existing images from the app.

### PR DESCRIPTION
**Pull request by jeffyzxc**
### All Submissions

* [ ] Can you post a screenshot of your changes (if applicable)?

### New Feature Submissions

* [ ] Does this relate to an existing issue, or has it been talked about prior to being done (if a major change)?

### Notes

Please note, your code must pass all tests and lint checks before it can be merged.

### PR Description
image
Screenshot 2023-05-15 203409
Screenshot 2023-05-15 203422
Screenshot 2023-05-15 203434
Screenshot 2023-05-15 203352

closes #1933 

### Testing

Has the code been tested, or does it need double checking?

### CODE REVIEW

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 1]
```diff
@@ -223,12 +224,22 @@ export default Vue.extend({
...
-        this.setSelectedCharacter(id);
-        return 'character-highlight';
+      const isCombatPage = (this as any).$router.currentRoute.path === '/combat';
+      const isTeamAdventureTab = (this as any).$route.query?.tab === COMBAT_TAB[COMBAT_TAB.TeamAdventure];
+      const fightMultiplier = Number(localStorage.getItem('fightMultiplier')) | 1;
+
+      if(isTeamAdventureTab && isCombatPage) {
+        const characterStamina = this.getCharacterStamina(id);
+
+        return characterStamina >= (fightMultiplier * 40) ?'character-highlight' : 'character';
```
Comment (Chriserus) [inline frontend/src/components/smart/CharacterDisplay.vue:234]: Here and in multiple places there seems to be a small formatting issue when it comes to spacing. Can we run some linter to fix these problems?

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 2]
```diff
@@ -273,8 +284,10 @@ export default Vue.extend({
     setActiveTab(tab: any) {
       (this as any).$router.push({ path: 'blacksmith', query: { tab: tab.route } }).catch(() => {});
       this.sideBarBlacksmith.forEach((sidebarTab: any) => {
-        if (sidebarTab.id === tab.id)
+        if (sidebarTab.id === tab.id) {
           sidebarTab.status = 'active';
+          console.log('setActiveTab');
```
Comment (Chriserus) [inline frontend/src/components/smart/CharacterDisplay.vue:289]: Unnecessary `console.log`

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 3]
```diff
@@ -293,7 +306,7 @@ export default Vue.extend({
           a.isSelected = true;
         }
         else {
-          a.isSelected = false;
+          a.isSelected = true;
```
Comment (Chriserus) [inline frontend/src/components/smart/CharacterDisplay.vue:309]: If both branches of this `if` statement do the same thing, shouldn't they be unified?

**Review by Chriserus — CHANGES_REQUESTED** [review]

File: frontend/src/components/smart/CharacterDisplay.vue [hunk 4]
```diff
@@ -223,12 +224,22 @@ export default Vue.extend({
       return await this.fetchIsCharacterRaiding({ characterID });
     },
     setListClassForSelChar(id: string, currentCharId: string): any {
-      if (id === currentCharId) {
-        this.setSelectedCharacter(id);
-        return 'character-highlight';
+      const isCombatPage = (this as any).$router.currentRoute.path === '/combat';
+      const isTeamAdventureTab = (this as any).$route.query?.tab === COMBAT_TAB[COMBAT_TAB.TeamAdventure];
+      const fightMultiplier = Number(localStorage.getItem('fightMultiplier')) | 1;
```
[the remark this answers is not in the document]
Comment (jeffyzxc) [reply frontend/src/components/smart/CharacterDisplay.vue:229]: should return 1

File: frontend/src/components/smart/CharacterDisplay.vue (hunk 1 above)
Comment (jeffyzxc) [reply frontend/src/components/smart/CharacterDisplay.vue:234]: I did, prolly lint ignore this as it didn't tag it as warning / error

**Pull request merged**